runtime/metricsview/executor_summary.go

```diff
@@ -0,0 +1,250 @@
+	}
+
+	// Build and execute the summary query
+	selectClauses := make([]string, 0, len(dimensions)*4)
+	var failedDimensions []string
+
+	for _, dim := range dimensions {
+		dimName := dim.Name
+		expr, err := e.olap.Dialect().MetricsViewDimensionExpression(dim)
+		if err != nil {
+			failedDimensions = append(failedDimensions, dimName)
+			continue // Skip this dimension instead of failing entirely
+		}
+
+		dialect := e.olap.Dialect()
+
+		// Add select clauses for min, max, has_nulls, and sample value
+		selectClauses = append(selectClauses,
+			fmt.Sprintf("MIN(%s) AS %s", expr, dialect.EscapeIdentifier(dimName+"__min")),
+			fmt.Sprintf("MAX(%s) AS %s", expr, dialect.EscapeIdentifier(dimName+"__max")),
+			fmt.Sprintf("COUNT(*) - COUNT(%s) > 0 AS %s", expr, dialect.EscapeIdentifier(dimName+"__has_nulls")),
+			fmt.Sprintf("ANY_VALUE(%s) AS %s", expr, dialect.EscapeIdentifier(dimName+"__sample")),
+		)
+	}
+
+	if len(selectClauses) == 0 {
+		if len(failedDimensions) > 0 {
+			return nil, fmt.Errorf("failed to get expressions for all dimensions: %v", failedDimensions)
+		}
+		return nil, fmt.Errorf("no dimensions to summarize")
+	}
+
+	timeDimExpr := e.olap.Dialect().EscapeIdentifier(e.metricsView.TimeDimension)
+	if e.metricsView.TimeDimension != "" {
+		for _, dim := range e.metricsView.Dimensions {
+			if dim.Name == e.metricsView.TimeDimension {
+				expr, err := e.olap.Dialect().MetricsViewDimensionExpression(dim)
+				if err == nil {
+					timeDimExpr = expr
+				}
+				break
+			}
+		}
+	}
+
+	// Use a recent time window with proper timestamp formatting
+	dimensionTimeRange := defaultTimeRange.Max.Add(-DefaultSampleTimeWindow)
+	whereClause := fmt.Sprintf("WHERE %s >= '%s'", timeDimExpr, dimensionTimeRange.Format(time.RFC3339))
+
+	escapedTableName := e.olap.Dialect().EscapeTable(e.metricsView.Database, e.metricsView.DatabaseSchema, e.metricsView.Table)
+	sql := fmt.Sprintf("SELECT %s FROM %s %s LIMIT %d",
+		strings.Join(selectClauses, ", "),
+		escapedTableName,
+		whereClause,
+		summaryQueryLimit,
+	)
+
+	rows, err := e.olap.Query(ctx, &drivers.Statement{
+		Query:            sql,
+		Priority:         e.priority,
+		ExecutionTimeout: defaultExecutionTimeout,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute dimension summary query: %w", err)
+	}
+	defer rows.Close()
+
+	if !rows.Next() {
+		return nil, fmt.Errorf("dimension summary query returned no results")
+	}
+
+	values := make([]interface{}, len(dimensions)*statsPerDimension)
+
+	// Create scan targets
+	for i := 0; i < len(values); i++ {
+		values[i] = new(interface{})
+	}
+
+	if err := rows.Scan(values...); err != nil {
+		return nil, fmt.Errorf("failed to scan dimension summary results: %w", err)
+	}
+
+	// Build the result summaries
+	summaries := make([]DimensionSummary, len(dimensions))
+
+	for i, dim := range dimensions {
+		dataType := getDimensionDataType(dim)
+
+		minValue, maxValue, sampleValue, hasNulls, err := extractDimensionStats(values, i)
```

Comment:
You know they'll exist, otherwise the logic is very broken and a panic is warranted. So would suggest just:
```suggestion
		minValue, maxValue, sampleValue, hasNulls := vals[i], vals[i+1], vals[i+2], vals[i+3]
```